instrumentation/httptrace/httptrace.go

```diff
@@ -112,16 +112,26 @@ func StartRequestSpan(r *http.Request, opts ...tracer.StartSpanOption) (*tracer.
 
 					ctx := r.Context()
 					spanctx.ForeachBaggageItem(func(k, v string) bool {
+						// always propagate into the context
 						ctx = baggage.Set(ctx, k, v)
+						// tag only if we allow all or it’s in our map
+						if cfg.allowAllBaggage || func() bool {
+							_, ok := cfg.baggageTagKeys[k]
+							return ok
+						}() {
+							ssCfg.Tags["baggage."+k] = v
+						}
```

Comment:
@rachelyangdog @mtoffl01 I didn't realize this was possible. I'd rather to promote the anonymous function to an unexported function to improve readability.

Reply:
I also didn't realize this was possible 😆